src/openlifu/db/database.py

```diff
@@ -206,6 +205,29 @@ def write_volume(self, subject, volume, on_conflict="error"):
 
         self.logger.info(f"Added volume with ID {volume_id} for subject {subject_id} to the database.")
 
+    def write_solution(self, session:Session, solution:Solution, on_conflict: OnConflictOpts="error"):
+        solution_ids = self.get_solution_ids(session.subject_id, session.id)
+
+        if solution.id in solution_ids:
+            if on_conflict == "error":
+                raise ValueError(f"Solution with ID {solution.id} already exists in the database.")
+            elif on_conflict == "overwrite":
+                self.logger.info(f"Overwriting solution with ID {solution.id} in the database.")
+            elif on_conflict == "skip":
+                self.logger.info(f"Skipping solution with ID {solution.id} as it already exists.")
+                return
+            else:
+                raise ValueError("Invalid 'on_conflict' option. Use 'error', 'overwrite', or 'skip'.")
```

Comment:
Use enum classes here for type safety instead of raw strings. If the goal is to only port the matlab code then please ignore this (since this is what was written in matlab).